scripts/apitypings/testdata/generics/generics.ts

```diff
@@ -1,41 +1,46 @@
+// Code generated by 'guts'. DO NOT EDIT.
+
+export type Comparable = string | number | boolean;
+
 // From codersdk/generics.go
-export interface Complex<C extends comparable, S extends Single, T extends Custom> {
-	readonly dynamic: Fields<C, boolean, string, S>;
-	readonly order: FieldsDiffOrder<C, string, S, T>;
-	readonly comparable: C;
-	readonly single: S;
-	readonly static: Static;
+export interface Complex<C extends Comparable, S extends Single, T extends Custom> {
+    readonly dynamic: Fields<C, boolean, string, S>;
+    readonly order: FieldsDiffOrder<C, string, S, T>;
+    readonly comparable: C;
+    readonly single: S;
+    readonly static: Static;
 }
 
 // From codersdk/generics.go
-export interface Dynamic<A extends any, S extends Single> {
-	readonly dynamic: Fields<boolean, A, string, S>;
-	readonly comparable: boolean;
-}
+export type Custom = string | boolean | number | number | string[] | (number | null);
```

Comment:
I don't think this is a huge deal, but the TypeScript output here looks a little bit jank. I'm guessing the `(number | null)` is from a nullable int in Go, but the type `number | (number | null)` collapses down to just `number | null`

Reply:
:+1: I can definitely unest these